### PR TITLE
User model and Donor Level model

### PR DESCRIPTION
**Pull request by anoshaminai**
r @yoninachmany  @weronica 
- Updated user model with new fields
- Created Donor Level model

User model is still waiting for zip code, region, and affiliation. Also need to update method to create fake data with new fields.

### CODE REVIEW

File: app/models/user.py [hunk 1]
```diff
@@ -44,6 +44,30 @@ def __repr__(self):
         return '<Role \'%s\'>' % self.name
 
 
+class UserLinks(db.Model):
+    __tablename__ = 'userlinks'
```
Comment (yoninachmany) [inline app/models/user.py]: `ZIPCode`'s `__table__name` was `zip_codes` --> change to `user_links`?

File: app/models/user.py [hunk 2]
```diff
@@ -44,6 +44,30 @@ def __repr__(self):
...
+    linkedin_link = db.Column(db.String(100))
+    twitter_link = db.Column(db.String(100))
+    instagram_link = db.Column(db.String(100))
+
+    def __repr__(self):
+        return '<UserLinks \'%i\'>' % self.id
+
+
+class DonorLevel(db.Model):
+    __tablename__ = 'donorlevels'
```
Comment (yoninachmany) [inline app/models/user.py]: `ZIPCode`'s `__table__name` was `zip_codes` --> change to `donor_levels`?

File: app/models/user.py [hunk 3]
```diff
@@ -53,9 +77,19 @@ class User(UserMixin, db.Model):
     email = db.Column(db.String(64), unique=True, index=True)
     password_hash = db.Column(db.String(128))
     role_id = db.Column(db.Integer, db.ForeignKey('roles.id'))
+
+    donor_level_id = db.Column(db.Integer, db.ForeignKey('donorlevels.id'))
```
Comment (yoninachmany) [inline app/models/user.py]: if we are changing name of table, should change to `ForeignKey('donor_levels.id')`

File: app/models/user.py [hunk 4]
```diff
@@ -53,9 +77,19 @@ class User(UserMixin, db.Model):
...
+    age = db.Column(db.Integer)
```
Comment (yoninachmany) [inline app/models/user.py]: Would it be preferable to change this field to a birthday so we can dynamically track someone's age over time?

Comment (skylerwharton) [inline app/models/user.py]: Or just birth year. I like this idea.

Comment (yoninachmany) [inline app/models/user.py]: down with birth year, thoughts @anoshaminai?

Comment (Mebsworth) [inline app/models/user.py]: I added birthday field to edit-profile so we can just remove this and I can add it with my changes.

Comment (anoshaminai) [inline app/models/user.py]: So @Mebsworth I should just take out age entirely?

File: app/models/user.py (hunk 2 above)
Comment (anoshaminai) [inline app/models/user.py]: I agree with these changes - shall do.

**Comment by anoshaminai** [comment]
Edited the table names as @yoninachmany suggested. Waiting to confirm on removing age field to make that change, other than that, anything else? @yoninachmany @weronica

**Comment by yoninachmany** [comment]
@anoshaminai once you get the go-ahead from @Mebsworth you're good to go!

File: app/models/user.py [hunk 5]
```diff
@@ -53,9 +77,18 @@ class User(UserMixin, db.Model):
...
     role_id = db.Column(db.Integer, db.ForeignKey('roles.id'))
+    donor_level_id = db.Column(db.Integer, db.ForeignKey('donor_levels.id'))
+    age = db.Column(db.Integer)
+    bio = db.Column(db.Text)
+    causes = db.Column(db.Text)
+    user_links = db.relationship('UserLinks', lazy='joined', uselist=False)
     resources = db.relationship('Resource', backref='user', lazy='dynamic')
     resource_reviews = db.relationship('ResourceReview', backref='user',
                                        lazy='dynamic')
+    # zip_code_id = db.Column(db.Integer, db.ForeignKey('ZIPCodes.id'))
```
Comment (aharelick) [inline app/models/user.py]: let's remove these four comment lines

Comment (anoshaminai) [inline app/models/user.py]: Done

Comment (anoshaminai) [inline app/models/user.py]: Done

**Comment by anoshaminai** [comment]
@aharelick Am I good to merge?

**Comment by aharelick** [comment]
send it to the cloud, :shipit:

**Pull request merged**